### PR TITLE
[vm] Cherry-picking multiple fixes for reliability to main (cherry-pick of #718)

### PR DESCRIPTION
**Pull request by wrwg**
* add limits to push on stack
* Define limit in the depth of dependencies in a module

Co-authored-by: Teng Zhang <rahxephon89@163.com>
Co-authored-by: Dario Russi <dario@aptoslabs.com>

### CODE REVIEW

**Review by rahxephon89 — APPROVED** [review]
LGTM, only one nit

File: language/move-vm/runtime/src/loader.rs [hunk 1]
```diff
@@ -1181,7 +1195,9 @@ impl Loader {
             &mut visited,
             &mut friends_discovered,
             allow_module_loading_failure,
+            0,
         )?;
+        // println!("({}) FRIENDS[{}]: {:#?}", dependencies_depth, friends_discovered.len(), friends_discovered);
```
Comment (dariorussi) [inline language/move-vm/runtime/src/loader.rs:1200]: can we remove this print please? that was my bad...

**Review by dariorussi — COMMENTED** [review]
it looks good to me, can you please remove the print as commented?

File: language/move-vm/runtime/src/loader.rs (hunk 1 above)
Comment (dariorussi) [reply language/move-vm/runtime/src/loader.rs:1200]: sorry did not see this went in already

**Pull request merged**